connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/aiagent/model/request/provider/GoogleVertexAiProviderConfiguration.java

```diff
@@ -0,0 +1,92 @@
+
+@TemplateSubType(id = GOOGLE_VERTEX_AI_ID, label = "Google Vertex AI")
+public record GoogleVertexAiProviderConfiguration(
+    @Valid @NotNull GoogleVertexAiConnection googleVertexAi) implements ProviderConfiguration {
+
+  @TemplateProperty(ignore = true)
+  public static final String GOOGLE_VERTEX_AI_ID = "google-vertex-ai";
+
+  public record GoogleVertexAiConnection(
+      @NotBlank
+          @TemplateProperty(
+              group = "provider",
+              label = "Project ID",
+              description = "Specify Google Cloud project ID",
+              type = TemplateProperty.PropertyType.String,
+              feel = Property.FeelMode.optional,
+              constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+          String projectId,
+      @NotBlank
+          @TemplateProperty(
+              group = "provider",
+              label = "Location",
+              description = "Specify the region where AI inference should take place",
+              type = TemplateProperty.PropertyType.String,
+              feel = Property.FeelMode.optional,
+              constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+          String location,
+      @Valid @NotNull GoogleVertexAiProviderConfiguration.GoogleVertexAiModel model) {}
+
+  public record GoogleVertexAiModel(
+      @NotBlank
+          @TemplateProperty(
+              group = "model",
+              label = "Model",
+              description =
+                  "Specify the model ID. Details in the <a href=\"https://cloud.google.com/vertex-ai/docs/generative-ai/models\" target=\"_blank\">documentation</a>.",
+              type = TemplateProperty.PropertyType.String,
+              feel = Property.FeelMode.optional,
+              constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+          String model,
+      @Valid GoogleVertexAiModelParameters parameters) {
+
+    public record GoogleVertexAiModelParameters(
+        @Min(0)
+            @TemplateProperty(
+                group = "model",
+                label = "Maximum tokens",
+                tooltip =
+                    "Maximum number of tokens that can be generated in the response. <br><br>Details in the <a href=\"https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference\" target=\"_blank\">documentation</a>.",
+                type = TemplateProperty.PropertyType.Number,
+                feel = Property.FeelMode.required,
+                optional = true)
+            Integer maxTokens,
+        @Min(0)
+            @TemplateProperty(
+                group = "model",
+                label = "Temperature",
+                tooltip =
+                    "Floating point number between 0 and 1. The higher the number, the more randomness will be injected into the response. <br><br>Details in the <a href=\"https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference\" target=\"_blank\">documentation</a>.",
```

Comment:
🔧 Unfortunately the range is provider specific and for Google it seems to be between 0 and 2, but can be model specific. We need at least adapt it to what is stated in the docs, but in doubt we can also remove the range and reference the documentation.

Reply:
It is actually between 0 and 1. The L4J doc is wrong. Since topP is a probability it cannot be more than 1. In the referenced documentation of Vertex AI the range is [0-1].

Reply:
I was referring to the temperature which [Google documents](https://cloud.google.com/vertex-ai/generative-ai/docs/model-reference/inference#request) as being between 0-2.

Reply:
Ah sorry I misread :man_facepalming: